### PR TITLE
Add tip for when returning `null` from `createToken()`

### PR DESCRIPTION
**Pull request by jeroenseegers**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | #5264 |

### CODE REVIEW

File: cookbook/security/api_key_authentication.rst [hunk 1]
```diff
@@ -28,7 +28,9 @@ allows you to implement such a scheme really easily.
 
 Your exact situation may differ, but in this example, a token is read
 from an ``apikey`` query parameter, the proper username is loaded from that
-value and then a User object is created::
+value and then a User object is created:
+
+.. code-block:: php
```
Comment (xabbuh) [inline cookbook/security/api_key_authentication.rst]: This change is not needed. We always use a double colon when we want to end a sentence with a colon followed by a PHP code block.

Comment (xabbuh) [inline cookbook/security/api_key_authentication.rst]: same below

Comment (jeroenseegers) [inline cookbook/security/api_key_authentication.rst]: Ah, ok. Reason I changed it everywhere is that the `code-block` notation was being used in the `Handling Authentication Failure` section at line 294 in this document. Should that be changed to a double colon instead then?

Comment (xabbuh) [inline cookbook/security/api_key_authentication.rst]: That example actually is a bit different as the sentence before the code block ends with a period.

Comment (jeroenseegers) [inline cookbook/security/api_key_authentication.rst]: Ok, I removed the addition of the `code-block` statements.

File: cookbook/security/api_key_authentication.rst [hunk 2]
```diff
@@ -128,6 +128,11 @@ will cause authentication to fail. You might want to return ``null`` instead
 to just skip the authentication, so Symfony can fallback to another authentication
 method, if any.
 
+.. tip::
```
Comment (WouterJ) [inline cookbook/security/api_key_authentication.rst]: I think a caution is more suited for the content of the block.

**Comment by WouterJ** [comment]
:+1:

**Comment by jeroenseegers** [comment]
I changed the `.tip::`-block to a `.caution::`-block.

**Comment by xabbuh** [comment]
:+1:

**Comment by xabbuh** [comment]
Thank you @jeroenseegers. I have merged your changes into the `2.7` branch from which they will be merged up to the other maintained branches. That's why your pull request is shown as closed instead of merged.

**Comment by jeroenseegers** [comment]
Great, thanks! :+1:

**Comment by WouterJ** [comment]
On second thought, unless I'm wrong of course, we're telling a wrong story here. Now it looks like you need to use *the anonymous* authentiction provider, otherwise things won't work. However, you need just *some* authentication provider that is executed after your custom one in order to work.

`null` is like: skip, let's pass this request to the next one. If you do enable the anonymous authentication provider, you indeed get an AnonymousToken. But you can also have another auth provider that is run after yours and you get another token.

Short story: Just make sure you have another authentication provider enabled if you return `null`. This is what the last sentence of the paragraph tries to tell us: "*You might want to return ``null`` instead to just skip the authentication, so Symfony can fallback to another authentication method, if any.*"

Btw, @jeroenseegers, this is nothing against your contribution, that's still great!

**Pull request closed without merging**